src/event_handler.rs

```diff
@@ -21,17 +21,83 @@ impl LipaEventHandler {
+                    } => {
+                        info!(
+                            "Registered incoming invoice payment for {} msat with hash {:?}",
+                            amount_msat, payment_hash
+                        );
+                        *payment_preimage
+                    }
+                    PaymentPurpose::SpontaneousPayment(preimage) => {
+                        info!(
+                            "Registered incoming spontaneous payment for {} msat with hash {:?}",
+                            amount_msat, payment_hash
+                        );
+
+                        Some(*preimage)
+                    }
+                };
+
+                match payment_preimage {
+                    Some(preimage) => {
+                        self.channel_manager.claim_funds(preimage);
+                    }
+                    None => {
+                        error!("Failed to claim funds for payment with hash {:?}: No preimage. Failing HTLC backwards", payment_hash);
+                        self.channel_manager.fail_htlc_backwards(payment_hash);
+                    }
+                }
+            }
+            Event::PaymentClaimed {
+                payment_hash,
+                amount_msat,
+                purpose: _,
+            } => {
+                info!(
+                    "Claimed payment for {} msat with hash {:?}",
+                    amount_msat, payment_hash
+                );
+            }
             Event::PaymentSent { .. } => {}
             Event::PaymentFailed { .. } => {}
             Event::PaymentPathSuccessful { .. } => {}
             Event::PaymentPathFailed { .. } => {}
             Event::ProbeSuccessful { .. } => {}
             Event::ProbeFailed { .. } => {}
-            Event::PendingHTLCsForwardable { .. } => {}
+            Event::PendingHTLCsForwardable {
+                time_forwardable: _,
+            } => {
+                // The variable time_forwardable is meant to be used to obfuscate the timing
+                // of when a payment is being forwarded/accepted.
+                // For the time being (while Lipa is the only LSP for these wallets)
+                // this measure may only obfuscate for routing nodes on the payment path,
+                // whether a payment that is being routed through the Lipa-LSP
+                // is being routed towards a Lipa user or towards a third party node.
+                // Since the Lipa users don't forward payments anyways,
+                // the Lipa-LSP itself will always know the destination of the payment anyways.
+                // For this reason, we don't deem this timely obfuscation to provide
+                // any meaningful privacy advantage for our case and therefore
+                // do not delay the acceptance of HTLC using the time_forwardable variable for now.
+
+                self.channel_manager.process_pending_htlc_forwards();
+            }
```

Comment:
Please check out b84fb1e0a9ba5bd638b861380c9d0881a85ea889 .

This event is actually being triggered when the node receives a payment. Without it the payment flow doesn't work. Check out the GitHub Actions on the last commit on https://github.com/getlipa/lipa-lightning-lib/tree/debug/temporary-purposefully-failing-branch for proof.

Reply:
So what you can see on there is that the test never terminates at all, because `self.channel_manager.process_pending_htlc_forwards();` triggers the event `PaymentReceived`.

Reply:
I misunderstood the event. Thank you for verifying!